tests/lib/preseed.sh

```diff
@@ -43,17 +43,33 @@ umount_ubuntu_image() {
 setup_preseeding() {
     local IMAGE_MOUNTPOINT=$1
     local CORE_IMAGE
+    local SNAPD_IMAGE
 
-    # TODO: on 20.04 there is no core_*.snap anymore, just snapd
     CORE_IMAGE=$(find "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/snaps/" -name "core_*.snap")
-    unsquashfs "$CORE_IMAGE"
-    cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
-    # XXX to satisfy version check; this will go away once preseeding
-    # is available in 2.44
-    echo "VERSION=2.44.0" > squashfs-root/usr/lib/snapd/info
-    rm "$CORE_IMAGE"
-    #shellcheck source=tests/lib/snaps.sh
-    . "$TESTSLIB"/snaps.sh
-    mksnap_fast squashfs-root "$CORE_IMAGE"
-    sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e 's/^(\s+)name: core/\1name: core\n\1unasserted: true/'
+    if [ -e "$CORE_IMAGE" ]; then
+        unsquashfs "$CORE_IMAGE"
+        cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
+        # XXX to satisfy version check; this will go away once preseeding
+        # is available in 2.44
+        echo "VERSION=2.44.0" > squashfs-root/usr/lib/snapd/info
+        rm "$CORE_IMAGE"
+        #shellcheck source=tests/lib/snaps.sh
+        . "$TESTSLIB"/snaps.sh
+        mksnap_fast squashfs-root "$CORE_IMAGE"
+        sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e 's/^(\s+)name: core/\1name: core\n\1unasserted: true/'
+    fi
+
+    SNAPD_IMAGE=$(find "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/snaps/" -name "snapd_*.snap")
+    if [ -e "$SNAPD_IMAGE" ]; then
+        unsquashfs "$SNAPD_IMAGE"
+        cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
+        # XXX to satisfy version check; this will go away once preseeding
+        # is available in 2.44
+        echo "VERSION=2.44.0" > squashfs-root/usr/lib/snapd/info
+        rm "$SNAPD_IMAGE"
+        #shellcheck source=tests/lib/snaps.sh
+        . "$TESTSLIB"/snaps.sh
+        mksnap_fast squashfs-root "$SNAPD_IMAGE"
+        sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e 's/^(\s+)name: snapd/\1name: snapd\n\1unasserted: true/'
+    fi
```

Comment:
These two blocks look pretty much identical.  Could you collapse them back down with a loop like `for snap in core snapd; do ...`?

Reply:
Yeah, sure thing.

Reply:
Done